docs/source/miscellaneous/doc/adr/0014-cassandra-targeted-sweep.md

```diff
@@ -0,0 +1,283 @@
+to enable better parallelisation of targeted sweep, the maximum number of **Shards** that is supported is *256*, effectively
+splitting the queue into that number of disjoint queues. Note that once that the number of shards is increased, it cannot be
+lowered again.
+
+#### sweepableCells
+
+  This table stores the actual information about all the writes into AtlasDB. As a transaction is about to be committed, but before
+  data is persisted to the KVS, all writes from the transaction are partitioned into shards and also according to the sweep strategy
+  of the table. Then, if there are 50 or fewer writes in a partition, we persist the necessary information in the same number of cells
+  in a **non-dedicated row**, with the row components calculated as described below. If there are more than 50 writes in a
+  partition, we only insert a single cell to the non-dedicated row acting as a reference to one or more **dedicated rows** which are
+  used exclusively for the writes of this transaction and partition.
+
+  **Row components**:
+  - **timestamp_partition**: a `VAR_LONG` derived from the start timestamp of the writes in that row. For *non-dedicated* rows, this
+    is the fine timestamp partition. For *dedicated rows*, this is the start timestamp itself.
+  - **metadata**, a 4 byte `BLOB` encoding of a `TargetedSweepMetadata` as follows:
+    - 1 bit for *sweep strategy*: 0 for thorough, and 1 for conservative.
+    - 1 bit marking if this is a *dedicated row*: 0 for non-dedicated, 1 for dedicated.
+    - 8 bits for *shard number*, between 0 and 255.
+    - 6 bits for use by dedicated rows, marking its ordinal number, between 0 and 63.
+    - 16 bits unused for now.
+
+  **Column components**:
+  - **timestamp_modulus**: a `VAR_LONG` storing the start timestamp of the write modulo 50_000.
+  - **write_index**: a `VAR_SIGNED_LONG` whose purpose is overloaded. In case we are storing 50 or fewer writes into the same row,
+    the *write_index* is a non-negative increasing number used to deduplicate multiple writes with the same start timestamp. If we
+    are storing more writes, we instead only put a single cell in the non-dedicated row acting as a reference to one or more
+    dedicated rows. In this case, the *write_index* is a negative number between 1 and 64 indicating how many dedicated rows we are
+    using (each dedicated row can contain up to *100_000* cells). In dedicated rows, the *write_index* is once again used for deduplication since all entries in the row will have the same
+    *timestamp_modulus*.
+
+  **Value**: a `WriteReference` containing the remaining required metadata for targeted sweep: the `TableReference`, `Cell`, and
+    a `boolean` specifying if the write was a tombstone.
+
+  Since a persisted size of a `Cell` object can be up to 3_000 bytes (dominating the size of the above entry), and we can have at
```

Comment:
Should check this - the 1000 byte limit might be on the entire cell length rather than the row/column names separately

Reply:
Might also be worth prefacing this section by saying that the parameters were chosen so as to not produce excessively large rows.

Reply:
It checks that each is at most 1500